### PR TITLE
CORE-1249: changes to support obtaining stat information from listings.

### PR DESCRIPTION
**Pull request by slr71**
I also updated some dependencies and added some functions to make it easier to test this library from the REPL.

### CODE REVIEW

File: src/clj_icat_direct/queries.clj [hunk 1]
```diff
@@ -194,17 +195,17 @@
   [dirname basename group-ids-query]
   [(str "WITH "
      "objs AS ("
-       "SELECT c.coll_id AS object_id, 'collection' AS type, c.coll_name AS full_path, regexp_replace(c.coll_name, '.*/', '') AS base_name, 0 as data_size, c.create_ts, c.modify_ts FROM r_coll_main c WHERE coll_name = ? || '/' || ? "
+       "SELECT c.coll_id AS object_id, 'collection' AS type, c.coll_name AS full_path, regexp_replace(c.coll_name, '.*/', '') AS base_name, 0 as data_size, c.create_ts, c.modify_ts, NULL AS data_checksum FROM r_coll_main c WHERE coll_name = ? || '/' || ? "
```
Comment (ianmcorvidae) [inline src/clj_icat_direct/queries.clj:198]: (applies several spots, choosing one at random)

wasn't sure, and you may have tested this already, but I'm wondering if adding this column will break schema validation for any listings. We might need to either filter it down to the supported keys (but for all I remember we might already) or add the checksum to the listings (which might be the better option, but more out of scope)

File: src/clj_icat_direct/queries.clj [hunk 2]
```diff
@@ -294,6 +297,23 @@
   [uuid]
   (mk-paths-for-uuids [uuid]))
 
+(defn mk-perms-for-item
+  [path]
+  (let [[dirname basename] ((juxt #(.getParent %) #(.getName %)) (file path))]
+    (sql/format
+     (-> {:with [[:object-lookup {:union [(-> (h/select [:coll_id :object_id])
```
Comment (ianmcorvidae) [inline src/clj_icat_direct/queries.clj]: doesn't need to be now, partly noting for myself but it'd apply to anyone who wants to do the refactor, it'd be nice to pull this object-lookup CTE code into a helper function and use it in at least `mk-get-item` (which needs more columns than this, but that could be parameterized). I'd like to refactor the listing stuff to use a strategy more like this too, so it'd apply there too eventually

(also, I like us using honeysql for this stuff. lots cleaner :+1: )

**Review by ianmcorvidae — APPROVED** [review]
All looks pretty reasonable to me, a comment and a small question but fine to ship I think.

File: src/clj_icat_direct/queries.clj (hunk 1 above)
Comment (slr71) [reply src/clj_icat_direct/queries.clj:198]: I spot checked a few places, and I think that we're filtering the list of fields down in most places. Admittedly, I haven't done an exhaustive search. I'll make sure to hit all of the endpoints before merging the changes. If I do have to make changes, I'll probably put them in data-info or terrain.

File: src/clj_icat_direct/queries.clj (hunk 2 above)
Comment (slr71) [reply src/clj_icat_direct/queries.clj]: I like the idea of splitting this into a separate function. I'll probably do that now. I agree about HoneySQL. It's a lot easier to deal with than long unwieldy string constants.

File: src/clj_icat_direct/queries.clj (hunk 1 above)
Comment (ianmcorvidae) [reply src/clj_icat_direct/queries.clj:198]: Sounds good to me. I think data-info (or maybe terrain) would definitely be the right place to do it, glad to have it added in here.

**Comment by slr71** [comment]
Thanks for the review! I'm going to merge this now.

**Pull request merged**